core/consensus/babe/impl/babe_config_repository_impl.cpp

```diff
@@ -111,24 +115,34 @@ namespace kagome::consensus::babe {
       warp(lock, finalized);
     }
 
-    auto genesis_res = config({block_tree_->getGenesisBlockHash(), 0}, false);
-    if (not genesis_res) {
-      SL_ERROR(logger_, "get config at genesis error: {}", genesis_res.error());
-      return false;
+    if (!timings_) {
+      auto genesis_res = config({block_tree_->getGenesisBlockHash(), 0}, false);
+      if (genesis_res.has_value()) {
+        auto &genesis = genesis_res.value();
+        timings_.init(genesis->slot_duration, genesis->epoch_length);
+        SL_DEBUG(logger_,
+                 "Timing was initialized: slot is {}ms, epoch is {} slots",
+                 timings_.slot_duration.count(),
+                 timings_.epoch_length);
+      }
     }
-    auto &genesis = genesis_res.value();
-    slot_duration_ = genesis->slot_duration;
-    epoch_length_ = genesis->epoch_length;
 
+    [[maybe_unused]] bool active_ = false;
+
+    // TODO Perhaps, should be observed all non-finalized blocks
```

Comment:
Not clear message. Maybe you meant `TODO Perhaps, should be observed by all non-finalized blocks`?